src/app/shared/components/carriage/carriage.component.ts

```diff
@@ -0,0 +1,135 @@
+/*
+** How to use in parents component **
+
+In the parent component, you can get data for user-selected seats form CarriageComponent:
+Method for example:
+
+onSelectedSeatsChange(seatsData: CarriageSelectedSeats) {
+    console.log('Выбранные места:', seatsData);
+  }
+
+In HTML template:
+ <app-carriage
+    [code]="carriage.code"
+    [name]="carriage.name"
+    [rows]="carriage.rows"
+    [leftSeats]="carriage.leftSeats"
+    [rightSeats]="carriage.rightSeats"
+    [carriageNumber]="carriage.carriageNumber"
+    [occupiedSeat]="carriage.occupiedSeat"
+    (selectedSeatsChange)="onSelectedSeatsChange($event)"
+  ></app-carriage>
+*/
+
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+export enum SeatStatus {
+  FREE = 'free',
+  OCCUPIED = 'occupied',
+  BOOKED = 'booked',
+}
+export interface Seat {
+  numberInCarriage: number;
+  status: SeatStatus.FREE | SeatStatus.OCCUPIED | SeatStatus.BOOKED;
+}
+
+export interface CarriageSelectedSeats {
+  carriageNumber: number;
+  carriageCode: string;
+  carriageName: string;
+  selectedSeats: Seat[];
+}
+
+@Component({
+  selector: 'app-carriage',
+  templateUrl: './carriage.component.html',
+  styleUrl: './carriage.component.scss',
+})
+export class CarriageComponent implements OnInit {
+  // code - auto-generated unique code for item. User to update carriage.
+  @Input() code!: string;
+
+  // name - unique name of the carriage type.
+  @Input() name!: string;
+
+  // rows - number of rows
+  @Input() rows!: number;
+
+  // leftSeats - - the number of seats to the left of the aisle in a row (2 in example below)
+  @Input() leftSeats!: number;
+
+  // rightSeats - the number of seats to the right of the aisle in a row (3 in example below)
+  @Input() rightSeats!: number;
+
+  // should be calculated by the index of the carriage in the train+1
+  @Input() carriageNumber!: number;
```

Comment:
Add required flag for every necessary input

Reply:
Thank you, I didn't know about that possibility!

Reply:
Done